### PR TITLE
rados: add GetConn to return connection used for IOContext

### PR DESCRIPTION
**Pull request by clwluvw**
This can help to shutdown the connection used for the IOContext

Signed-off-by: Seena Fallah <seenafallah@gmail.com>

## Checklist
- [x] Added tests for features and functional changes
- [x] Public functions and types are documented
- [x] Standard formatting is applied to Go code
- [ ] Is this a new API? Is this new API marked PREVIEW?

### CODE REVIEW

**Comment by clwluvw** [comment]
@phlogistonjohn This can also fix #424 because if the `conn` variable didn't store anywhere and I think it will be deleted by the garbage collector and so the connections will be broken! (I checked it with `ss` to list the connections and didn't find any connection to my Ceph cluster). After I store it in my struct then the connections won't break!

**Comment by phlogistonjohn** [comment]
This is similar to my response on #661. However since you say this fixes #424 I'm a bit more interested, if we can lose the new `GetConn()` function. What do you think?

**Comment by clwluvw** [comment]
I think this really sounds like your style of programming so doesn't matter so much but keeping the `conn` in `ioctx` as I mentioned in the issue and also the way it could be a problem, that makes sense to keep it. Are you ok with that?

**Comment by clwluvw** [comment]
> > I think this really sounds like your style of programming so doesn't matter so much but keeping the `conn` in `ioctx` as I mentioned in the issue and also the way it could be a problem, that makes sense to keep it. Are you ok with that?
> 
> Could you rephrase that? I'm not really sure, what you want to say, sorry.

I meant are you ok with the part to keep the `conn` into the `ioctx` because of the issue I've mentioned above and @phlogistonjohn also was agree on that?

**Comment by phlogistonjohn** [comment]
I'm going to decline to take the PR and close it.
The aspect of these changes that holds a reference to the Conn in IOContexts has been merged as part of PR #713. Since the only remaining part of the pr was a public method to access the Conn from the iocontext wasn't something we were inclined to take before there's not much more to do with this PR.  Please note that the issue causing the crashes should be fixed now and will be included in the next release. I hope you understand.
